src/viewport.cpp

```diff
@@ -1023,6 +1024,10 @@ static TileHighlightType GetTileHighlightType(TileIndex t)
 		if (IsTileType(t, MP_STATION) && GetStationIndex(t) == _viewport_highlight_station->index) return THT_WHITE;
 		if (_viewport_highlight_station->TileIsInCatchment(t)) return THT_BLUE;
 	}
+	if (_viewport_highlight_station_rect != nullptr) {
+		const StationRect *r = &_viewport_highlight_station_rect->rect;
+		if (r->PtInExtendedRect(TileX(t), TileY(t))) return THT_BLUE;
+	}
```

Comment:
Feels a bit cramped without line breaks. 😉 

```suggestion
	
	if (_viewport_highlight_station_rect != nullptr) {
		const StationRect *r = &_viewport_highlight_station_rect->rect;
		if (r->PtInExtendedRect(TileX(t), TileY(t))) return THT_BLUE;
	}
	
```